pkg/cluster/deploystorage_resources.go

```diff
@@ -96,6 +97,16 @@ func (m *manager) storageAccount(name, region string, encrypted bool) *arm.Resou
 		},
 	}
 
+	// when installing via Hive we need to allow Hive to persist the installConfig graph in the cluster's storage account
+	// TODO: add AKS shard support
+	hiveShard := 1
+	if m.installViaHive && strings.Index(name, "cluster") == 0 {
+		virtualNetworkRules = append(virtualNetworkRules, mgmtstorage.VirtualNetworkRule{
+			VirtualNetworkResourceID: to.StringPtr("/subscriptions/" + m.env.SubscriptionID() + "/resourceGroups/" + m.env.ResourceGroup() + fmt.Sprintf("/providers/Microsoft.Network/virtualNetworks/aks-net/subnets/ClusterSubnet-%03d", hiveShard)),
```

Comment:
nit: can we use a formatted string for the entire resourceID instead of a combination of string concatenation and string format?

Reply:
+1, especially since we are already using a formatted string at the end

Reply:
Good call, +1 for readability. Done.